Added Start Scene

Milind21 opened the pull request:
-Added start menu as a new scene.
-Triggers customize menu which can be used to enter the scene.

Milind21 committed:
Added Start Scene

-Added start menu as a new scene.
-Triggers customize menu which can be used to enter the scene.

Milind21 commented:
@blst15ki can you take a look at the wait logic? I have added a wait time which just works for the first time. Apart from that the menu works as expected

blst15ki commented on Assets/Scripts/MainMenu.cs:
```diff
@@ -0,0 +1,89 @@
+    }
+    void Update()
+    {
+        if (customizeMenuScript.wait)
```
When running in the start scene, it throws a NullReferenceException since customizeMenuScript in the Main Menu script is not set in the Start scene.

blst15ki commented on Assets/Scenes/Start.unity:
Could you add Garden Sim as the title?

blst15ki commented on Assets/Scripts/CustomizeMenu.cs:
```diff
@@ -0,0 +1,100 @@
+        XInput = "js2";
+        AInput = "js10";
+        button = 0;
+        // imagea1.color = Color.yellow; // By default selection on avatar 1
```
Should be kept (highlight first option to avoid confusing player), though change it to what is actually the first option in the menu

blst15ki commented:
Sorry I was clicking the wrong button, meant to post altogether as a single review instead of multiple single comments.

blst15ki commented on Assets/Scripts/MainMenu.cs:
```diff
@@ -0,0 +1,89 @@
+        XInput = "js2";
+        AInput = "js10";
+        button = 0;
+        // image0.color = Color.yellow;
```
Should also be kept to avoid confusing player (let them know what button is first)

blst15ki reviewed (changes requested):
The real reason why both menus were working at the same time is because you don't have only one menu script running at a time. For example, at the start, both scripts are active so when you hit X to cycle the options, the CycleButton method is called for both scripts. In this case, I don't think you need to set a wait boolean variable.

Milind21 commented:
- disable and enable the scripts too 
- remove the bools

Milind21 committed:
Addressed changes and added background

- Rectified highlighting of button
- Rectified quit and back interference
- Added title (Garden Sim)
- Added background

blst15ki reviewed (changes requested):
Start scene looks good in the editor now.
- Could you add the source of the background to the source file?
- The start screen needs to be linked to a camera (does not show up on Android)
- Can you resize the elements to fit closer together? For the main menu, the title is barely visible and the exit button is at the center. For the customize menu, the avatar options go off the screen.

Milind21 commented:
1) Is this related to screen overlay and screen camera? 
2) If so will the normal main camera work or do I have to setup the basic character from assignment 1?

blst15ki commented:
> 1. Is this related to screen overlay and screen camera?
> 2. If so will the normal main camera work or do I have to setup the basic character from assignment 1?

1. Yes, it is related. The canvas needs to be on screen camera.
2. Yes, the current main camera will work.

Milind21 committed:
Changed from screen overlay to screen camera

- Added camera to screen  camera mode
- Adjusted the buttons so they appear on android scene

Milind21 committed:
Updated src.txt

- Added background image (grass) for the start scene
